src/main/scala/com/high-performance-spark-examples/ml/CustomPipeline.scala

```diff
@@ -122,7 +122,10 @@ class SimpleIndexer(override val uid: String)
     import dataset.sparkSession.implicits._
     val words = dataset.select(dataset($(inputCol)).as[String]).distinct
       .collect()
-    new SimpleIndexerModel(uid, words)
+    val model = new SimpleIndexerModel(uid, words)
+    model.set(inputCol, $(inputCol))
```

Comment:
That was my first thought as well. (Actually, I tried using `copy`, not `copyValues`, but I think it would have the same problem.) Both `copy` and `copyValues` seem to need `extra` which is of type `ParamMap` but the `paramMap` property is private and `extractParamMap` just returns an empty map.